### PR TITLE
Add RAM tests, bug fixes

### PR DESCRIPTION
**Pull request by ryley-o**
## Description of the change

Add RAM tests and fix any bugs found.

All changes after audit start are meant to be captured on this branch, and eventually folded into the `ram-minter` branch and PR.

### CODE REVIEW

File: packages/contracts/contracts/minter-suite/Minters/MinterRAMV0.sol [hunk 1]
```diff
@@ -1185,8 +1185,37 @@ contract MinterRAMV0 is ReentrancyGuard, ISharedMinterV0, ISharedMinterRAMV0 {
...
-     * on this minter contract.
+     * @notice Convenience view function that returns the bid value associated
+     * with a given slot index for the specified project's auction, in Wei.
+     * Reverts if the slot index is out of range (greater than 511).
+     * Reverts if the project's auction is not configured.
+     * @param projectId Project ID to get the bid value for
+     * @param coreContract Core contract address for the given project
+     * @param slotIndex Slot index to get the bid value for
+     */
+    function slotIndexToBidValue(
```
Comment (ryley-o) [inline packages/contracts/contracts/minter-suite/Minters/MinterRAMV0.sol:1196]: new function does not require audit, and is for convenience only (may be removed due to bytecode size limit)

File: packages/contracts/contracts/libs/v0.8.x/minter-libs/RAMLib.sol [hunk 2]
```diff
@@ -1560,12 +1570,15 @@ library RAMLib {
...
+            // initialize min and max bid slot index metadata
+            RAMProjectConfig_.minBidSlotIndex = slotIndex;
+            RAMProjectConfig_.maxBidSlotIndex = slotIndex;
         }
         // require at least one token allowed in auction
         // @dev this case would revert in _removeMinBid, but prefer clean error
         // message here
         uint256 numTokensInAuction = RAMProjectConfig_.numTokensInAuction;
-        require(numTokensInAuction > 0, "No bids in auction");
+        require(numTokensInAuction > 0, "No tokens in auction");
```
Comment (ryley-o) [inline packages/contracts/contracts/libs/v0.8.x/minter-libs/RAMLib.sol:1581]: improved error message clarity

File: packages/contracts/test/minter-suite-minters/RAM/MinterRAM/integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,3185 @@
...
+        // not in State B error
+        await expectRevert(
+          config.minter
+            .connect(config.accounts.user)
+            .createBid(config.projectZero, config.genArt721Core.address, 0, {
+              value: config.basePrice,
+            }),
+          revertMessages.onlyLiveAuction
+        );
+        // configure, remian in State A
```
Comment (lyaunzbe) [inline packages/contracts/test/minter-suite-minters/RAM/MinterRAM/integration.test.ts:200]: spelling nit: `remain`

Comment (lindsgil) [reply packages/contracts/test/minter-suite-minters/RAM/MinterRAM/integration.test.ts]: 🙏

**Comment by ryley-o** [comment]
went ahead and merged this to consolidate everything into the ram-minter branch for simplicity

**Pull request merged**